src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            linkHead(value);
+        } else {
+            linkTail(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0) {
+            throw new IndexOutOfBoundsException("The index out of bounds!");
+        }
+        if (size == 0 || index == 0) {
+            linkHead(value);
+        } else if (index == size) {
+            linkTail(value);
+        } else {
+            Node<T> currentNode = getElement(index);
+            Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
+            size++;
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getElement(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue = getElement(index).value;
+        getElement(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> current = getElement(index);
+        final T value = current.value;
+        if (index == 0 && size == 1) {
+            current = null;
+            head = current;
+        } else if (index == 0 && size > 1) {
+            current.next.prev = null;
+            head = current.next;
+        } else if (index > 0 && index < size - 1) {
+            current.next.prev = current.prev;
+            current.prev.next = current.next;
+        } else if (index > 0 && index == size - 1) {
+            current.prev.next = null;
+            tail = current.prev;
+        }
+        size--;
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = getIndex(object);
```

Comment:
lets better get node by value and call unlink method
because you are iterating over the list a few times